### PR TITLE
Added Explanations

### PR DESCRIPTION
**Pull request by TrifanBogdan24**
Here are some changes I made and my approach of Tour of Rust:
- a few words about what is Rust and its unique aspects
- working in terminal with Rust (compiling, running and cargo commands) 
- macros
-  detailed information about data types and collections
- self and Self in terms of implementing a static / non-static method for a struct
- how to work with cargo

### CODE REVIEW

**Comment by TrifanBogdan24** [comment]
Solved issues:
- #431 (https://github.com/richardanaya/tour_of_rust/issues/431#issue-1167495785)
- #423 (https://github.com/richardanaya/tour_of_rust/issues/423)
- #306 (https://github.com/richardanaya/tour_of_rust/issues/306)
- #289 (https://github.com/richardanaya/tour_of_rust/issues/289)

Added features:
- a few words about Rust and its unique aspects
- terminal commands (compiling, running a Rust executable, cargo)
- detailed information about data types and collections
- how to use cargo
- the differences between a library and a binary crate
- pointer

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Delete the `.idea` folder from the repository.

**Comment by alexandruradovici** [comment]
Split this in two pull requests, one that fixes the existing chapters and one that adds the extra chapter.

File: lessons/en/chapter_4.yaml [hunk 1]
```diff
@@ -181,6 +181,17 @@
...
   code: >-
     https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=fn%20do_something_that_might_fail(i%3A%20i32)%20-%3E%20Result%3Cf32%2C%20String%3E%20%7B%0A%20%20%20%20if%20i%20%3D%3D%2042%20%7B%0A%20%20%20%20%20%20%20%20Ok(13.0)%0A%20%20%20%20%7D%20else%20%7B%0A%20%20%20%20%20%20%20%20Err(String%3A%3Afrom(%22this%20is%20not%20the%20right%20number%22))%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20-%3E%20Result%3C()%2C%20String%3E%20%7B%0A%20%20%20%20%2F%2F%20concise%20but%20assumptive%20and%20gets%20ugly%20fast%0A%20%20%20%20let%20v%20%3D%20do_something_that_might_fail(42).unwrap()%3B%0A%20%20%20%20println!(%22found%20%7B%7D%22%2C%20v)%3B%0A%20%20%20%20%0A%20%20%20%20%2F%2F%20this%20will%20panic!%0A%20%20%20%20let%20v%20%3D%20do_something_that_might_fail(1).unwrap()%3B%0A%20%20%20%20println!(%22found%20%7B%7D%22%2C%20v)%3B%0A%20%20%20%20%0A%20%20%20%20Ok(())%0A%7D%0A
+- title: About panic!
+  content_markdown: >
+    In Rust, `panic!` is a macro used to stop the execution of the program
+    without a recoverable error. When a panic occurs, the program immediately
+    stop, unwinding the stack and cleaning up resources along the way.
```
Comment (i-alexa) [inline lessons/en/chapter_4.yaml]: stops*

File: lessons/en/chapter_9.yaml [hunk 2]
```diff
@@ -130,6 +254,26 @@
...
+    
+    The Rust `use` is equivalent to Python's `import` if we are to make
+    a comparison between these two programming languages.
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=%2F%2F+Bringing+the+%60println%60+function+into+scope%0Ause+std%3A%3Aio%3A%3Aprintln%3B%0A%0Afn+main%28%29+%7B%0A++++println%21%28%22Hello%2C+World%21%22%29%3B%0A%7D%0A
+- title: extern crate
+  content_markdown: >
+    The `extern crate` directive was used in older versions of Rust (before the 2018 edition).
+    
+    Now, is no longer required to use `extern crate` in code.
```
Comment (i-alexa) [inline lessons/en/chapter_9.yaml]: it is no longer*

**Review by iulianalexa — CHANGES_REQUESTED** [review]

**Comment by TrifanBogdan24** [comment]
The changes are split in the following two different Pull Request:
- https://github.com/UPB-CS-OpenSourceUpstream/tour_of_rust/pull/19 on branch `fixing`
- https://github.com/UPB-CS-OpenSourceUpstream/tour_of_rust/pull/20 on branch `new-chapter`

**Pull request closed without merging**